### PR TITLE
Adds options for mounts

### LINKED ISSUE #195: Enhancement: Support `options` field on `/sys/mounts`
Right now, there's no way of performing certain logic based on if the `kv` endpoint is 1 or 2. 

I think this is just a case of adding 

```ruby
# @!attribute [r] options
    #   Options of the mount.
    #   @return [String]
    field :options
```

to the `lib/vault/api/sys/mount.rb` file

### PR DESCRIPTION
**Pull request by petems**
* Allows us to check what kv version is mounted
* Also allows us to retrieve options from existing 
mounts
* Restricts to 0.10+ for specs

Closes #195

### CODE REVIEW

File: lib/vault/api/sys/mount.rb [hunk 1]
```diff
@@ -44,8 +49,10 @@ def mounts
     #   the type of mount
     # @param [String] description
     #   a human-friendly description (optional)
-    def mount(path, type, description = nil)
-      payload = { type: type }
+    # @param [String] options
+    # mount type specific options that are passed to the backend (optional)
+    def mount(path, type, description = nil, options = {})
```
Comment (grosser) [inline lib/vault/api/sys/mount.rb]: prefer that options includes `:description` ... maybe some fallback `if options is not a hash, then treat it as description`

File: lib/vault/api/sys/mount.rb [hunk 2]
```diff
@@ -44,8 +49,10 @@ def mounts
     #   the type of mount
     # @param [String] description
     #   a human-friendly description (optional)
-    def mount(path, type, description = nil)
-      payload = { type: type }
+    # @param [String] options
```
Comment (grosser) [inline lib/vault/api/sys/mount.rb]:
```suggestion
    # @param [Hash] options
```

File: lib/vault/api/sys/mount.rb [hunk 3]
```diff
@@ -44,8 +49,10 @@ def mounts
...
+    # mount type specific options that are passed to the backend (optional)
```
Comment (grosser) [inline lib/vault/api/sys/mount.rb]:
```suggestion
    #   mount type specific options that are passed to the backend (optional)
```

File: lib/vault/api/sys/mount.rb [hunk 4]
```diff
@@ -44,8 +49,10 @@ def mounts
...
+    def mount(path, type, description = nil, options = {})
+      payload = options.merge({ type: type })
```
Comment (grosser) [inline lib/vault/api/sys/mount.rb]:
```suggestion
      payload = options.merge(type: type)
```

File: spec/integration/api/sys/mount_spec.rb [hunk 5]
```diff
@@ -12,6 +12,15 @@ module Vault
         expect(sys.type).to eq("system")
         expect(sys.description).to include("system endpoints")
       end
+
+      it "shows options for mounts" do
+        next unless has_options_for_mount?
```
Comment (grosser) [inline spec/integration/api/sys/mount_spec.rb]: could use skip so it's marked as skipped
```suggestion
        skip unless has_options_for_mount?
```

File: spec/integration/api/sys/mount_spec.rb [hunk 6]
```diff
@@ -12,6 +12,15 @@ module Vault
...
+
+        kv = subject.mounts[:kv]
+        expect(kv).to be_a(Mount)
+        expect(kv.type).to eq("kv")
+        expect(kv.options).to eq(:version => 2)
```
Comment (grosser) [inline spec/integration/api/sys/mount_spec.rb]:
```suggestion
        expect(kv.options).to eq(version: 2)
```

File: spec/integration/api/sys/mount_spec.rb [hunk 7]
```diff
@@ -22,6 +31,24 @@ module Vault
         expect(result.type).to eq("aws")
         expect(result.description).to eq("")
       end
+
+      it "returns the options from a mount" do
+        next unless has_options_for_mount?
+
+        expect(subject.mount("kv_v2_mount", "kv", 'kv v2 mount', options: {version: "2"})).to be(true)
```
Comment (grosser) [inline spec/integration/api/sys/mount_spec.rb]: isn't this supposed to be 
```suggestion
        expect(subject.mount("kv_v2_mount", "kv", 'kv v2 mount', version: "2")).to be(true)
```

Comment (petems) [reply spec/integration/api/sys/mount_spec.rb]: Nope, it's an options hash containing the entry of version . AFAIK :https://www.vaultproject.io/api/system/mounts.html#options

File: lib/vault/api/sys/mount.rb (hunk 1 above)
Comment (petems) [reply lib/vault/api/sys/mount.rb]: Not sure I understand this comment, do you mean that options contains a description? Or to remove the extra attribute?

Comment (grosser) [reply lib/vault/api/sys/mount.rb]: both :D 

```ruby
def mount(path, type, options = {})
  description = (options.is_a?(Hash) ? options[:description] : options)
```

**Comment by maxadamo** [comment]
will anyone push the merge button? :smiley_cat:

**Pull request closed without merging**